File: barricade/src/main/java/com/mutualmobile/barricade/utils/AndroidAssetFileManager.java
/*
 * Copyright (c) 2011,2012,2013,2014,2015 Mutual Mobile. All rights reserved.
 */

package com.mutualmobile.barricade.utils;

import android.content.Context;
import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.InputStream;
import java.util.logging.Logger;

/**
 * Implementation of {@link AssetFileManager} to allow reading files from assets folder in Android
 */
public class AndroidAssetFileManager implements AssetFileManager {
  private Context mApplicationContext;
  private static final String TAG = "AndroidAssetFileManager";

  public AndroidAssetFileManager(Context applicationContext) {
    mApplicationContext = applicationContext;
  }

  public String getContentsOfFileAsString(String fileName) {
    try {
      InputStream is = getContentsOfFileAsStream(fileName);
      ByteArrayOutputStream bos = new ByteArrayOutputStream();
      int aByte = is.read();
      while (aByte != -1) {
        bos.write(aByte);
        aByte = is.read();
      }
      return bos.toString("UTF-8");
    } catch (IOException e) {
      Logger.getLogger(TAG).severe(e.getMessage());
    }
    return null;
  }

  @Override public InputStream getContentsOfFileAsStream(String fileName) {
    try {
      return mApplicationContext.getAssets().open(String.format("%s", fileName));
    } catch (IOException e) {
      Logger.getLogger(TAG).severe(e.getMessage());
    }
    return null;
  }
}

File: barricade/src/main/java/com/mutualmobile/barricade/utils/AndroidAssetFileManager.kt
package com.mutualmobile.barricade.utils

import android.content.Context
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.util.logging.Logger

/**
 * Implementation of [AssetFileManager] to allow reading files from assets folder in Android
 */
class AndroidAssetFileManager(private val applicationContext: Context) : AssetFileManager {
    private companion object {
        const val TAG = "AndroidAssetFileManager"
    }

    override fun getContentsOfFileAsString(fileName: String): String? {
        return try {
            val inputStream = getContentsOfFileAsStream(fileName)
            BufferedReader(inputStream?.reader()).readText()
        } catch (e: IOException) {
            Logger.getLogger(TAG).severe(e.message)
            null
        }
    }

    override fun getContentsOfFileAsStream(fileName: String): InputStream? {
        return try {
            applicationContext.assets.open(String.format("%s", fileName))
        } catch (e: IOException) {
            Logger.getLogger(TAG).severe(e.message)
            null
        }
    }
}
